src/features/events/components/EventsList/parts/NavigationFilter/index.tsx

```diff
@@ -34,7 +31,7 @@ export const NavigationFilter: FC<Props> = ({ onChange }) => (
 
     <MobileToggleLabel>
       <span>Show</span>
-      <select>
+      <select onChange={(e) => onChange(e.target.value as FilterType)}>
```

Comment:
This line is worth noting. As I am still new to typescript I didn't know this kind of type assertion is possible. When I had to solve this `onChange` event and I had the code `onChange(e.target.value)` my typescript was complaining that `string` cannot be assigned as `FilterType`. I tried to solve it by reverse mapping the `FliterType` `enum` object, which is not possible to do for `string` enums, so I ended up writing an onChnageHandler which mapped the `strings` to `FilterTypes`.
Nevertheless this solution is much more simpler.